src/librawspeed/decoders/Cr2Decoder.cpp

```diff
@@ -273,11 +273,10 @@ RawImage Cr2Decoder::decodeNewFormat() {
 
 RawImage Cr2Decoder::decodeRawInternal() {
   try {
-    if (hints.find("old_format") != hints.end()) {
+    if (hints.find("old_format") != hints.end())
       return decodeOldFormat();
-    } else {
-      return decodeNewFormat();
-    }
+
+    return decodeNewFormat();
```

Comment:
my 2c. : While I can see situations where an else after a return is not helpful, like an early error check and return or throw on failure, I think it is not generally a bad decision to allow else after return. In this example (and a few others) I find it a lot more readable with the else, since it makes it more obvious that those two options are "of the same category". It clearly communicates an "either A or B" intention by having both A and B on same indentation level.